internal/component/otelcol/receiver/tcplog/tcplog.go

```diff
@@ -0,0 +1,175 @@
+	LineEndPattern   string `alloy:"line_end_pattern,attr,optional"`
+	OmitPattern      bool   `alloy:"omit_pattern,attr,optional"`
+}
+
+func (c *MultilineConfig) Convert() *split.Config {
+	if c == nil {
+		return nil
+	}
+
+	return &split.Config{
+		LineStartPattern: c.LineStartPattern,
+		LineEndPattern:   c.LineEndPattern,
+		OmitPattern:      c.OmitPattern,
+	}
+}
+
+var _ receiver.Arguments = Arguments{}
+
+// SetToDefault implements syntax.Defaulter, providing default values.
+func (args *Arguments) SetToDefault() {
+	*args = Arguments{
+		Output: &otelcol.ConsumerArguments{},
+	}
+	args.DebugMetrics.SetToDefault()
+	args.ConsumerRetry.SetToDefault()
+}
+
+// Convert implements receiver.Arguments, converting these Arguments
+// into an OpenTelemetry Collector config object for the tcplogreceiver.
+func (args Arguments) Convert() (otelcomponent.Config, error) {
+	c := stanzainputtcp.NewConfig()
+	c.BaseConfig = stanzainputtcp.BaseConfig{
+		MaxLogSize:      helper.ByteSize(args.MaxLogSize),
+		ListenAddress:   args.ListenAddress,
+		TLS:             args.TLS.Convert(),
+		AddAttributes:   args.AddAttributes,
+		OneLogPerPacket: args.OneLogPerPacket,
+		Encoding:        args.Encoding,
+	}
+	if c.MaxLogSize == 0 {
+		c.MaxLogSize = tcpDefaultMaxLogSize
+	}
+	split := args.MultilineConfig.Convert()
+	if split != nil {
+		c.SplitConfig = *split
+	}
+
+	def := tcplogreceiver.ReceiverType{}.CreateDefaultConfig()
+	cfg := def.(*tcplogreceiver.TCPLogConfig)
+	cfg.InputConfig = *c
+
+	// consumerretry package is stanza internal so we can't just Convert
+	cfg.RetryOnFailure.Enabled = args.ConsumerRetry.Enabled
+	cfg.RetryOnFailure.InitialInterval = args.ConsumerRetry.InitialInterval
+	cfg.RetryOnFailure.MaxInterval = args.ConsumerRetry.MaxInterval
+	cfg.RetryOnFailure.MaxElapsedTime = args.ConsumerRetry.MaxElapsedTime
+
+	return cfg, nil
+}
+
+// Extensions implements receiver.Arguments, returning any needed extensions.
+func (args Arguments) Extensions() map[otelcomponent.ID]otelextension.Extension {
+	return nil
+}
+
+// Exporters implements receiver.Arguments, returning exporters by signal type.
+// This wrapper doesn't add any exporters internally; it defers to Output.
+func (args Arguments) Exporters() map[pipeline.Signal]map[otelcomponent.ID]otelcomponent.Component {
+	return nil
+}
+
+// NextConsumers implements receiver.Arguments, returning the next consumer pipeline.
+func (args Arguments) NextConsumers() *otelcol.ConsumerArguments {
+	return args.Output
+}
+
+// Validate implements syntax.Validator for basic argument checks.
+func (args *Arguments) Validate() error {
+	var errs error
+
+	if err := validateListenAddress(args.ListenAddress, "listen_address"); err != nil {
+		errs = multierror.Append(errs, err)
+	}
+
+	_, err := decode.LookupEncoding(args.Encoding)
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("invalid encoding: %w", err))
+	}
+
+	if args.MaxLogSize != 0 && (int64(args.MaxLogSize) < int64(minMaxLogSize)) {
```

Comment:
Do we need the `args.MaxLogSize != 0` check? I think the defaults should have already been applied by now. And if someone sets the value to 0 on purpose, they should get an error.